core/src/main/scala/kafka/server/AbstractFetcherThread.scala

```diff
@@ -680,12 +680,15 @@ abstract class AbstractFetcherThread(name: String,
        */
       val offsetAndEpoch = leader.fetchEarliestOffset(topicPartition, currentLeaderEpoch)
       val leaderStartOffset = offsetAndEpoch.offset
-      warn(s"Reset fetch offset for partition $topicPartition from $replicaEndOffset to current " +
-        s"leader's start offset $leaderStartOffset")
       val offsetToFetch = Math.max(leaderStartOffset, replicaEndOffset)
+      info(s"Reset fetch offset for partition $topicPartition from $replicaEndOffset to " +
```

Comment:
Why are we changing the log level? Moreover, by saying “new fetch offset”, we lose the intention which is to reset to the leader start offset, no?

Reply:
> Why are we changing the log level? Moreover, by saying “new fetch offset”, we lost the intention which is to reset to the leader start offset, no?

As discussed with kamalcph, if came here, there may be an unclean leader election occurs, which need to be logged as info, because no data truncated and resume from last fetch offset. the other case is need truncate fully and start from leader's log start offset, which belongs to a "abnormal" case. when executing the line log, we can not decide which case belong to. Or we may only need to output warn log for "abnormal" case.

Resetting to leader's log start offset is only a possible case, so we can ignore the intention of log start offset, because we have added a warn log(start from log start offset for this case.) before executing truncateFullyAndStartAt (log start offset).

Reply:
I see. In this case, would it make sense to move this log to the else branch of the if at L687? It is a bit weird to log this and then potentially the warning.

If we do this, we could also rewrite it as follow:
```
Reset fetch offset for partition $topicPartition from $replicaEndOffset to the current local replica's end offset $offsetToFetch
```

Would this make sense?

Reply:
> I see. In this case, would it make sense to move this log to the else branch of the if at L687? It is a bit weird to log this and then potentially the warning.
> 
> If we do this, we could also rewrite it as follow:
> 
> ```
> Reset fetch offset for partition $topicPartition from $replicaEndOffset to the current local replica's end offset $offsetToFetch
> ```
> 
> Would this make sense?

Thanks, I think it makes more sense this way. @dajac

Reply:
Updated according to your comment.